numpy/testing/utils.py

```diff
@@ -689,20 +689,13 @@ def chk_same_position(x_id, y_id, hasval='nan'):
 
         if isnumber(x) and isnumber(y):
             x_isnan, y_isnan = isnan(x), isnan(y)
-            x_isinf, y_isinf = isinf(x), isinf(y)
 
             # Validate that the special values are in the same place
             if any(x_isnan) or any(y_isnan):
                 chk_same_position(x_isnan, y_isnan, hasval='nan')
-            if any(x_isinf) or any(y_isinf):
-                # Check +inf and -inf separately, since they are different
-                chk_same_position(x == +inf, y == +inf, hasval='+inf')
-                chk_same_position(x == -inf, y == -inf, hasval='-inf')
 
             # Combine all the special values
             x_id, y_id = x_isnan, y_isnan
```

Comment:
Why not just `x_isnan | x_isinf, y_isnan | y_isinf`? I would find that just a little less confusing than trying to figure out what the next two lines do.